defi/src/api2/cron-task/dimensions.ts

```diff
@@ -63,14 +63,402 @@ const timeData = {
+  const days = Object.keys(records || {}).sort()
+  for (const timeS of days) {
+    const rec = records[timeS] || {}
+    const bl = rec.bl?.[recordType]  ?? rec.breakdownLabel?.[recordType]
+    if (!bl) continue
+    const ts = timeSToUnix(timeS)
+
+    const byLabel: any = {}
+
+    Object.entries(bl).forEach(([label, v]: any) => {
+      const n = +v || 0
+      byLabel[label] = (byLabel[label] ?? 0) + n
+    })
+
+    labelSeries.push([ts, byLabel])
+  }
+  return { labelSeries }
+}
+
+type MQ = { monthly: Record<string, number>; quarterly: Record<string, number> }
+type MQByLabel = { monthly: Record<string, Record<string, number>>, quarterly: Record<string, Record<string, number>> }
+type AggregatedProtocolOut = { adapterType: string, slug: string, metrics: Record<string, MQ | MQByLabel>, metadata?: Record<string, any> }
+
+function monthKeyFromUnix(ts: number) {
+  const d = new Date(ts * 1000)
+  const y = d.getUTCFullYear()
+  const m = String(d.getUTCMonth() + 1).padStart(2, '0')
+  return `${y}-${m}`
+}
+
+function quarterKeyFromUnix(ts: number) {
+  const d = new Date(ts * 1000)
+  const y = d.getUTCFullYear()
+  const q = Math.floor(d.getUTCMonth() / 3) + 1
+  return `${y}-Q${q}`
+}
+
+function rollupFromChart(chart: Array<[number, number]>): MQ {
+  const monthly: Record<string, number> = {}
+  const quarterly: Record<string, number> = {}
+  for (const [ts, val] of chart || []) {
+    if (typeof ts !== 'number' || typeof val !== 'number') continue
+    const mk = monthKeyFromUnix(ts)
+    const qk = quarterKeyFromUnix(ts)
+    monthly[mk] = (monthly[mk] ?? 0) + val
+    quarterly[qk] = (quarterly[qk] ?? 0) + val
+  }
+  return { monthly, quarterly }
+}
+
+function rollupFromLabelChart(series: Array<[number, Record<string, number>]>): MQByLabel {
+  const monthly: Record<string, Record<string, number>> = {}
+  const quarterly: Record<string, Record<string, number>> = {}
+  for (const [ts, labels] of series || []) {
+    if (typeof ts !== 'number' || typeof labels !== 'object') continue
+    const mk = monthKeyFromUnix(ts)
+    const qk = quarterKeyFromUnix(ts)
+    for (const [label, v] of Object.entries(labels || {})) {
+      const val = typeof v === 'number' ? v : Number(v) || 0
+      if (!monthly[label]) monthly[label] = {}
+      if (!quarterly[label]) quarterly[label] = {}
+      monthly[label][mk] = (monthly[label][mk] ?? 0) + val
+      quarterly[label][qk] = (quarterly[label][qk] ?? 0) + val
+    }
+  }
+  return { monthly, quarterly }
+}
+
+function readJSONSafe(fp: string): any | null {
+  try { return JSON.parse(fs.readFileSync(fp, 'utf-8')) } catch { return null }
+}
+
+function slugFromAllFile(filename: string) {
+  return filename.endsWith('all') ? filename.slice(0, -3) : filename
+}
+
+function slugFromBlFile(filename: string) {
+  if (filename.endsWith('-bl')) return filename.slice(0, -3)
+  if (filename.endsWith('bl')) return filename.slice(0, -2)
+  return filename
+}
+
+function findBuildDir(): string | null {
```

Comment:
no, this is the wrong approach, if you want to read a generated file, you should use `readRouteFile`

Reply:
*readRouteData